File: 1term/10.22/2/2/stack.cpp
#include <stdio.h>
#include "stack.h"

struct Stack
{
	ListElement *head;
};

struct ListElement
{
	ElementType value;
	ListElement *next;
};

Stack *create()
{
	Stack *newStack = new Stack;
	newStack->head = NULL;
	return newStack;
}

void addElement(Stack *stack, ElementType value)
{
	ListElement *newListElement = new ListElement;
	newListElement->value = value;
	newListElement->next = stack->head;
	stack->head = newListElement;
}

ElementType valueHead(Stack *stack)
{
	return stack->head->value;
}

void deleteElement(Stack *stack)
{
	if (stack->head == NULL)
		return;
	ListElement *temp = stack->head;
	stack->head = stack->head->next;
	delete temp;
}

void deleteStack(Stack *stack)
{
	while (stack->head != NULL)
		deleteElement(stack);
	delete stack;
}

bool isEmpty(Stack *stack)
{
	return (stack->head == NULL);
}

File: 2term/03.19/1calculator/1calculator/Stack.cs
using System;

namespace _1calculator
{
    public class Stack
    {
        public Stack()
        {
            head = null;
        }

        // add element newValue to head.
        public void Push(int newValue)
        {
            ListElement temp = new ListElement(newValue, head);
            head = temp;
        }

        // delete element in head and return value.
        public int Pop()
        {
            if (head == null)
                throw new StackException("Stack is empty. Can't pop");
            int val = head.Value();
            head = head.Next();
            return val; 
        }

        public bool IsEmpty()
        {
            return (head == null);
        }

        // print stack.
        public string Print()
        {
            string str = null;
            if (head == null)
                return str;
            str = head.Value().ToString();
            if (head.Next() != null)
                if (head.Next().Next() != null)
                    str = head.Next().Next().Value().ToString() + (char)head.Next().Value() + head.Value().ToString();
                else
                    str = head.Next().Value().ToString() + (char)head.Value();
            else
                str = head.Value().ToString();
            return str;
        }

       private ListElement head;

       private class ListElement
       {
           public ListElement()
           { }

           public ListElement(int value, ListElement next)
           {
               this.value = value;
               this.next = next;
           }

           public int Value()
           {
               return this.value;
           }

           public ListElement Next()
           {
               return this.next;
           }

           private int value;
           private ListElement next;
       }

    }
}
